app/main.py

```diff
@@ -39,51 +48,257 @@ def execute(self):
+                        "nodes": topology_update["nodes"],
+                        "links": topology_update["links"],
+                        }
+            else:
+                self.sdx_topology = topology_mock.topology_mock()
+            evaluate_topology = self.validate_sdx_topology()
+            if evaluate_topology["status_code"] == 200:
+                result = self.post_sdx_lc(event_type)
+                return result
+            return {"result": evaluate_topology['result'],
+                    "status_code": evaluate_topology['status_code']}
+        except Exception as err:  # pylint: disable=W0703
+            log.info(err)
+        return {"result": "No SDX Topology loaded", "status_code": 401}
+
     @listen_to("kytos/topology.*")
-    def listen_topology(self, event=None):
+    def listen_event(self, event=None):
         """Function meant for listen topology"""
-        f_name = " listen_topology "
-        log.info(f"{HSH}{f_name}get kytos topology {HSH}")
-        event_info = {}  # pylint: disable=W0201
-        if event:
-            allowed_events = [
-                "kytos/topology.switch.enabled",
-                "kytos/topology.switch.disabled",
-                "kytos/topology.link_up",
-                "kytos/topology.link_down",
-            ]
-            if event.name in allowed_events:
-                try:
-                    topology = requests.get(settings.KYTOS_TOPOLOGY_URL).json()
-                    log.info(
-                        f"{HSH}{f_name}constructor post request {HSH}"
-                    )
-                    event_info = {
-                        "event": event,
-                        "kytostopology": topology,
-                    }  # pylint: disable=W0201
-                    try:
-                        requests.post(settings.SDX_CONSTRUCTOR).json(event_info)
-                    except Exception as err:  # pylint: disable=W0703
-                        log.info(f"{HSH}{f_name}constructor post error {HSH}")
-                        log.info(err, event_info)
-                except Exception as err:  # pylint: disable=W0703
-                    log.info(
-                            f"{HSH}{f_name}get kytos topology error {HSH}"
-                    )
-                    log.info(err)
-        return event_info
-
-    @rest("v2/listen_topology", methods=["GET"])
-    def get_listen_topology(self):
-        """consume call listen Topology"""
-        f_name = " get_listen_topology "
+        f_name = " listen_event "
+        log.info(f"{HSH}{HSH}{HSH}{HSH}{HSH}")
+        log.info(f"{HSH}{f_name} {HSH}")
+        log.info(f"{HSH} dir(event): {dir(event)} {HSH}")
+        log.info(f"{HSH} event: {event} {HSH}")
+        log.info(f"{HSH} event.name: {event.name} {HSH}")
+        log.info(f"{HSH} event.content: {event.content} {HSH}")
+        log.info(f"{HSH}{HSH}{HSH}{HSH}{HSH}")
+        if event is not None and self.get_kytos_topology():
```

Comment:
`self.get_kytos_topology()` can result in a lot of IO here (when a higher frequency event is being published). I don't understand why a `GET` topology is necessary here, especially since the events it's already driven from topology. Also, it's calling `self.get_kytos_topology()` again in the end of this handler and returning, but notice that a decorated `@listen_to` future result is discarded so you actually won't get anything from that return [which another method is trying to call it, `listen_to` decorated handlers isn't meant to be called directly](https://github.com/atlanticwave-sdx/kytos-sdx-topology/blob/luisdev/app/main.py#L282), so another extra request which the contents aren't actually won't be used. 

Let me now what the intention was. Looks like what matters is what's being posted to sdx, but the calls to topology don't look optimal or necessary to me, correct me if misunderstood your intention here.

Reply:
I agree. This was in the initial code of the parse topology and is not necessary, I will get rid of this, thank you